### PR TITLE
Vin/flyers v2

### PR DESCRIPTION
**Pull request by vinnie4k**
## Overview
This PR migrates community board to AppDev’s backend as well as minor flyer changes.

### Flyers

- Rewrote the model for `Flyer` to match with the backend
- Wrote new GraphQl queries to fetch flyers
- Removed old network requests from Vin’s backend to AppDev’s backend
- Added sharing for flyers. Currently using the `flyerURL` (Instagram page)
- Used the `getTrendingArticlesQuery` for trending articles
- Added searching for flyers
- Refactored bookmarking and sharing for flyers

### Organizations

- Rewrote the model for `Organization` to match with the backend
- Wrote new GraphQl queries to fetch organizations

### Flyers Bookmarking

- Created a new icon on Figma for bookmarked flyers
- Wrote a new GraphQL query for bookmarking flyers
- Added bookmarked flyers to UserDefaults

## Test Coverage
- Used different devices with varying screen sizes in Simulator to see if there were any UI issues
- In-depth playtesting to minimize any bugs that may be present

## Next Steps
- Pagination for Flyers once we get a lot more flyers and organizations onboarded. Can’t really implement this until we have a lot of flyers on the backend.

## Screenshots

<details>
  <summary>Flyers</summary>
	<video src="https://github.com/cuappdev/volume-ios/assets/75594943/fd804ebb-8aa1-441b-85c3-9eebd70750ea" type="video/mp4">
</details>

### CODE REVIEW

File: Volume/View Models/TrendingViewModel.swift [hunk 1]
```diff
@@ -65,24 +65,12 @@ extension TrendingView {
...
-                        throw URLError(.badServerResponse)
-                    }
-                    return data
-                }
-                .decode(type: [Flyer].self, decoder: JSONDecoder.flyersDecoder)
-                .sink { completion in
-                    print("Fetching trending flyers: \(completion)")
-                } receiveValue: { [weak self] flyers in
-                    self?.flyers = flyers
+            Network.shared.publisher(for: GetTrendingFlyersQuery(limit: 2))
```
Comment (jjennifergu) [inline Volume/View Models/TrendingViewModel.swift:68]: why is the limit 2?

File: Volume/Models/Flyer.swift [hunk 2]
```diff
@@ -10,17 +10,48 @@ import Foundation
...
+    let endDate: Date
+    let flyerUrl: URL?
+    let imageUrl: URL?
+    let isTrending: Bool
     let location: String
+    let nsfw: Bool
     let organizations: [Organization]
-    let postURL: String
+    let organizationSlugs: [String]
+    let shoutouts: Int
```
[the remark this answers is not in the document]
Comment (vinnie4k) [reply Volume/Models/Flyer.swift:24]: Oh yea thanks for reminding me.

File: Volume/Models/Organization.swift [hunk 3]
```diff
@@ -8,14 +8,33 @@
...
+    
+    init(from organization: OrganizationFields) {
+        self.id = organization.id
+        self.backgroundImageUrl = URL(string: organization.backgroundImageUrl)
+        self.bio = organization.bio
+        self.categorySlug = OrganizationType(rawValue: organization.categorySlug) ?? .academic
+        self.name = organization.name
+        self.profileImageUrl = URL(string: organization.profileImageUrl)
+        self.slug = organization.slug
+        self.shoutouts = Int(organization.shoutouts)
```
[the remark this answers is not in the document]
Comment (vinnie4k) [reply Volume/Models/Organization.swift:31]: Yes. Only Flyers have `timesClicked`

File: Volume/View Models/TrendingViewModel.swift (hunk 1 above)
Comment (vinnie4k) [reply Volume/View Models/TrendingViewModel.swift:68]: The trending page only displays 2 flyers.

File: Volume/Views/Flyers/FlyerCellPast.swift [hunk 4]
```diff
@@ -171,10 +193,11 @@ struct FlyerCellPast: View {
...
     }
     
+    @ViewBuilder
     private var categoryType: some View {
-        Text(Organization.contentTypeString(
-                // TODO: May need to reimplement this once backend is finished
-                type: flyer.organizations[0].type)
+        if let categorySlug = flyer.organizations.first?.categorySlug {
+            Text(Organization.contentTypeString(
+                type: categorySlug)
```
[the remark this answers is not in the document]
Comment (vinnie4k) [reply Volume/Views/Flyers/FlyerCellPast.swift:200]: No. It just means that the category for that flyer is from the first organization. The organization type for all organizations on the same flyer should be the same, so it's fine to just take the first one.

**Pull request merged**